### PR TITLE
core: ardupilot mananger: asyncify router management, consume stderr/stdout

### LINKED ISSUE #2247: mavp2p produces a lot of stderr output, hanging after a while
[logfile_12-09-2023_19_46_48.log](https://github.com/bluerobotics/BlueOS/files/13625505/logfile_12-09-2023_19_46_48.log)

Then hanging is kinda of a different issue, though

### PR DESCRIPTION
**Pull request by Williangalvani**
Should help with mavp2p hanging after a while.

fix #2247

### CODE REVIEW

File: core/services/ardupilot_manager/mavlink_proxy/main.py [hunk 1]
```diff
@@ -45,18 +46,21 @@
...
-        time.sleep(1)
-    logger.info("Done.")
+    async def main() -> None:
+        tool = AbstractRouter.get_interface(args.tool)()
+        logger.info(f"Starting {tool.name()} version {tool.version()}.")
+
+        if tool:
+            manager.use(tool)
+        else:
+            logger.warning("No tool selected. Falling back to the first one found")
```
Comment (patrickelectric) [inline core/services/ardupilot_manager/mavlink_proxy/main.py:56]: We should say which one in the log

File: core/services/ardupilot_manager/mavlink_proxy/main.py [hunk 2]
```diff
@@ -45,18 +46,21 @@
...
-    logger.info(f"Command: {manager.command_line()}")
-    manager.start(args.master.split(":"))
-    while manager.is_running():
-        time.sleep(1)
-    logger.info("Done.")
+    async def main() -> None:
+        tool = AbstractRouter.get_interface(args.tool)()
+        logger.info(f"Starting {tool.name()} version {tool.version()}.")
+
+        if tool:
```
Comment (patrickelectric) [inline core/services/ardupilot_manager/mavlink_proxy/main.py:53]: Shouldn't this check be before the previous log message ?

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: core/services/ardupilot_manager/mavlink_proxy/main.py (hunk 2 above)
Comment (Williangalvani) [reply core/services/ardupilot_manager/mavlink_proxy/main.py:52]: good catch. fixed

File: core/services/ardupilot_manager/mavlink_proxy/main.py (hunk 1 above)
Comment (Williangalvani) [reply core/services/ardupilot_manager/mavlink_proxy/main.py]: done

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**